lib/src/working_copy.rs

```diff
@@ -612,8 +634,14 @@ impl TreeState {
                 // untracked
                 let file_type = new_file_state.file_type.clone();
                 self.file_states.insert(repo_path.clone(), new_file_state);
-                let file_value = self.write_path_to_store(&repo_path, &disk_path, file_type)?;
-                tree_builder.set(repo_path, file_value);
+                if let Some(file_value) = self.write_path_to_store(
+                    &repo_path,
+                    &disk_path,
+                    file_type,
+                    settings.max_new_file_size(),
+                )? {
+                    tree_builder.set(repo_path, file_value);
```

Comment:
I think it's better to check max size here since we're sure that the file is newly added. File size may change after checking, but I think that's okay.

Reply:
> File size may change after checking, but I think that's okay.

Yeah, this is impossible to avoid anyway; even the prior draft could race with appends performed after the fstat call.